tests/bluetooth/tester/src/btp_vcp.c

```diff
@@ -378,3 +535,423 @@ uint8_t tester_unregister_vocs(void)
+
+	bt_addr_le_copy(&ev.address, bt_conn_get_dst(conn));
+
+	ev.volume = volume;
+	ev.mute = mute;
+
+	tester_event(BTP_SERVICE_ID_VCP, BTP_VCP_STATE_EV, &ev, sizeof(ev));
+}
+
+static void btp_send_vcp_volume_flags_ev(struct bt_conn *conn, uint8_t flags)
+{
+	struct btp_vcp_volume_flags_ev ev;
+
+	bt_addr_le_copy(&ev.address, bt_conn_get_dst(conn));
+
+	ev.flags = flags;
+
+	tester_event(BTP_SERVICE_ID_VCP, BTP_VCP_FLAGS_EV, &ev, sizeof(ev));
+}
+
+static uint8_t vcp_supported_commands(const void *cmd, uint16_t cmd_len,
+				      void *rsp, uint16_t *rsp_len)
+{
+	struct btp_vcp_read_supported_commands_rp *rp = rsp;
+
+	/* octet 0 */
+	tester_set_bit(rp->data, BTP_VCP_READ_SUPPORTED_COMMANDS);
+	tester_set_bit(rp->data, BTP_VCP_VOL_CTLR_DISCOVER);
+	tester_set_bit(rp->data, BTP_VCP_VOL_CTLR_STATE_READ);
+	tester_set_bit(rp->data, BTP_VCP_VOL_CTLR_FLAGS_READ);
+	tester_set_bit(rp->data, BTP_VCP_VOL_CTLR_VOL_DOWN);
+	tester_set_bit(rp->data, BTP_VCP_VOL_CTLR_VOL_UP);
+	tester_set_bit(rp->data, BTP_VCP_VOL_CTLR_UNMUTE_VOL_DOWN);
+
+	/* octet 1 */
+	tester_set_bit(rp->data, BTP_VCP_VOL_CTLR_UNMUTE_VOL_UP);
+	tester_set_bit(rp->data, BTP_VCP_VOL_CTLR_SET_VOL);
+	tester_set_bit(rp->data, BTP_VCP_VOL_CTLR_UNMUTE);
+	tester_set_bit(rp->data, BTP_VCP_VOL_CTLR_MUTE);
+
+	*rsp_len = sizeof(*rp) + 1;
+
+	return BTP_STATUS_SUCCESS;
+}
+
+static void vcp_vol_ctlr_discover_cb(struct bt_vcp_vol_ctlr *vol_ctlr, int err, uint8_t vocs_count,
+				     uint8_t aics_count)
+{
+	struct bt_conn *conn;
+
+	if (err) {
+		LOG_DBG("Discovery failed (%d)", err);
+		return;
+	}
+
+	LOG_DBG("Discovery done with %u VOCS, %u AICS",
+		vocs_count, aics_count);
+
+	bt_vcp_vol_ctlr_conn_get(vol_ctlr, &conn);
+
+	if (bt_vcp_vol_ctlr_included_get(vol_ctlr, &included) != 0) {
+		LOG_DBG("Could not get included services");
+		memset(&chrc_handles.vocs_handles, 0, sizeof(chrc_handles.vocs_handles));
+		memset(&chrc_handles.aics_handles, 0, sizeof(chrc_handles.aics_handles));
+	} else {
+		aics_client_instance.aics_cnt = included.aics_cnt;
+		aics_client_instance.aics = included.aics;
+		bt_vocs_client_cb_register(vol_ctlr->vocs[0], &vocs_cl_cb);
+		bt_aics_client_cb_register(vol_ctlr->aics[0], &aics_client_cb);
+
+		struct bt_vocs_client *cli =
+			CONTAINER_OF(vol_ctlr->vocs[0], struct bt_vocs_client, vocs);
+
+		chrc_handles.vocs_handles.state = cli->state_handle;
+		chrc_handles.vocs_handles.location = cli->location_handle;
+		chrc_handles.vocs_handles.control = cli->control_handle;
+		chrc_handles.vocs_handles.desc = cli->desc_handle;
+		chrc_handles.aics_handles.state = vol_ctlr->aics[0]->cli.state_handle;
+		chrc_handles.aics_handles.gain = vol_ctlr->aics[0]->cli.gain_handle;
+		chrc_handles.aics_handles.type = vol_ctlr->aics[0]->cli.type_handle;
+		chrc_handles.aics_handles.status = vol_ctlr->aics[0]->cli.status_handle;
+		chrc_handles.aics_handles.control = vol_ctlr->aics[0]->cli.control_handle;
+		chrc_handles.aics_handles.desc = vol_ctlr->aics[0]->cli.desc_handle;
```

Comment:
How come vocs and AICS are being treated differently here?

Reply:
Fixed